File: runtime-java/CompilerDemo/Runtime.java
package CompilerDemo;

import java.util.Scanner;
import java.util.Locale;


public class Runtime {
  static {
    Locale.setDefault(Locale.ROOT);
  }

  public static String read() {
    Scanner scanner = new Scanner(System.in);
    return scanner.nextLine();
  }

  public static void print(String p0) {
    System.out.print(p0);
  }

  public static void println(String p0) {
    System.out.println(p0);
  }

  public static int toInt(String p0) {
    return Integer.parseInt(p0);
  }

  public static double toFloat(String p0) {
    return Double.parseDouble(p0);
  }

  public static String convert(int v) {
    return "" + v;
  }

  public static String convert(double v) {
    return "" + v;
  }

  public static String convert(boolean v) {
    return "" + v;
  }

  public static String concat(String a, String b) {
    return a + b;
  }

  public static int compare(String a, String b) {
    return a.compareTo(b);
  }
}

File: runtime-net/runtime.cs
using System;
using System.Globalization;
using System.Threading;


namespace CompilerDemo {

  class Runtime {
    static Runtime() {
      Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;
    }

    public static string read() {
      return Console.ReadLine();
    }

    public static void print(string p0) {
      Console.Write(p0);
    }

    public static void println(string p0) {
      Console.WriteLine(p0);
    }

    public static int toInt(string p0) {
      return Convert.ToInt32(p0);
    }

    public static double toFloat(string p0) {
      return Convert.ToDouble(p0);
    }

    public static string convert(int v) {
      return Convert.ToString(v);
    }

    public static string convert(double v) {
      return Convert.ToString(v);
    }

    public static string convert(bool v) {
      return Convert.ToString(v);
    }

    public static string concat(string a, string b) {
      return a + b;
    }

    public static int compare(string a, string b) {
      return a.CompareTo(b);
    }
  }
}
